app/workers/problem_series/import_worker.rb

```diff
@@ -49,7 +49,10 @@ def perform
               problem.problem_sets << problem_set
 
               # set problem weighting in official problem set
-              ProblemSetProblem.find_by(problem_set_id: problem_set.id, problem_id: problem.id).update_attributes(weighting: manager.get_maxscore)
+              # can't use problem.problem_set_associations.find_by(problem_set_id: ...) because it
+              # operates on the database rather than the cache, and the problem might not have been
+              # saved yet
+              problem.problem_set_associations.select{|assoc|assoc.problem_set_id == problem_set.id}.first.update_attributes(weighting: manager.get_maxscore)
```

Comment:
I'm not familiar with Rails, but I wonder if there is a more idiomatic way to express this.

Also, do we want to add some spaces in the closure?

```suggestion
              problem.problem_set_associations.select{ |assoc| assoc.problem_set_id == problem_set.id }.first.update_attributes(weighting: manager.get_maxscore)
```

Reply:
> I'm not familiar with Rails, but I wonder if there is a more idiomatic way to express this.

Improvements welcome, they can be done as separate commits.

> Also, do we want to add some spaces in the closure?

I added spaces in the new code (commit 48d2345), but not the rest of the file (e.g. [line 214](https://github.com/NZOI/nztrain/blob/76733803893c5e7d2565230fc2c950f01e3c79dd/app/workers/problem_series/import_worker.rb#L214)).